documentation/topics/glossary.md

```diff
@@ -0,0 +1,127 @@
+# Glossary
+
+## Action
+
+An action describes an operation that can be performed for a given resource; it is the _verb_ to a resource's _noun_. Examples of actions:
+
+- User._create_
+- Comment._delete_
+- BlogPost._publish_
+- Article._search_
+
+Ash supports four different types of actions - `create`, `read`, `update` and `destroy` (collectively often abbreviated as CRUD). A resource can define multiple actions per action type, eg. a _publish_ action would be considered an `update` because it is updating an existing instance of a resource. Actions are much more flexible than simple CRUD, but these four action types serve as templates for anything you might want to do.
+
+See the [Actions guide](/documentation/topics/actions.md) for more information.
+
+## Actor
+
+The entity that performs an action. Most actions are run on direct user request, eg. if a user presses a Create button on a page then the actor is the user; but an actor might also be an organization, a group, or the system itself.
+
+Actors are referenced during authorization - ensuring that the actor is allowed to perform an action, before it takes place. The actor can also be used within actions, to record which entity performed the action.
+
+See the [Security guide](/documentation/topics/security.md#actors) for more information.
+
+## Aggregate
+
+An aggregate is a special type of field for a resource, one that summarizes related information of the record. A more specialized type of a [calculation](#calculation).
+
+If a Project resource has_many Ticket resources, an example of an aggregate on the Project might be to count the tickets associated to each project.
+
+See the [Aggregates guide](/documentation/topics/aggregates.md) for more information.
+
+## API
+
+A method of broadly separating resources into different [bounded contexts](https://martinfowler.com/bliki/BoundedContext.html). Small apps might only have one API, in which case you can set-and-forget it, but apps with larger domains can benefit from different contexts having different views of the same resource.
+
+See `d:Ash.Api.Dsl` for more information.
+
+## Attribute
+
+A piece of data belonging to a resource. The most basic building block; an attribute has a type and a value and is stored within the context of a domain model.
+
+See `d:Ash.Resource.Dsl.attributes` for more information.
+
+## Calculation
+
+A calculation is a special type of field for a resource, one that is not directly stored in the data layer but generated on-demand when specifically requested as part of a query. Typically it will derive from other information on the record, but it may come from some other data source entirely.
+
+See the [Calculations guide](/documentation/topics/calculations.md) for more information.
+
+## Changeset
+
+Changesets encapsulate data changes made while creating or updating an instance of a resource. A wrapper around Ecto changesets, they include data validations and also have their own callback hook lifecycle.
```

Comment:
They aren't actually a wrapper around ecto changesets. AshPostgres creates an ecto changeset from an ash changeset before inserting but that is just an implementation detail.